sdcm/argus_test_run.py

```diff
@@ -170,6 +169,47 @@ def _prepare_docker_resource_setup(sct_config: SCTConfiguration):
     return cloud_setup
 
 
+def parse_run_details(details: list) -> tuple:
+    """
+    Parse a jenkins job url into a three element tuple containing
+    release, group, test names. Most runs should fit into this pattern easily
+    and some will need some additional parsing / adjustments
+    Examples:
+    scylla-master/releng-testing/artifacts/artifacts-ami-test
+    -> [scylla-master, releng-testing-artifacts, artifacts-ami-test]
+    scylla-master/artifacts/artifacts-amazon2-test
+    -> [scylla-master, artifacts, artifacts-amazon2-test]
+    manager-master/ubuntu20-sanity-test
+    -> [manager-master, manager-tests, ubuntu20-sanity-test]
+    """
+    release = ""
+    group = ""
+    name = ""
+    if "manager" in details[0]:
+        release, *name = details
+        name = "-".join(name)
+        group = "manager-tests"
+    elif "scylla-operator" in details[0]:
+        _, release, group, *name = details
+        name = "-".join(name)
+    elif "scylla-staging" in details[0]:
+        release, group, *name_split = details
+        name = "-".join(name_split)
+    elif len(details) == 3:
+        release, group, name = details
+    elif len(details) < 3:
+        LOGGER.warning("Non-parseable build name detected, this run won't be stored in argus: %s", details)
+    else:
+        # try to make up a test name from arbitrary nesting
+        release, *group, name = details
+        group = "-".join(group)
```

Comment:
Amazingly good candidate for using `match` operator :) Do we have plans to backport it to pre-3.10 branches?